src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Persistence/Orchestration/BundleOrchestratorOperationTests.cs

```diff
@@ -107,6 +105,56 @@ public void GivenABatchOperation_WhenAppendedMultipleResourcesInParallelWaitForA
             Assert.Equal(numberOfResources, operation.CurrentExpectedNumberOfResources);
         }
 
+        [Fact]
+        public async Task GivenABatchOperation_WhenJustOneResourcedIsAppendedAndAllOtherResourcesAreHanging_ThenCancelTheOperationAfter120SecondsToAvoidRunningForever()
+        {
+            // Long running test.
+
+            // A security exit clause in Bundle Orchestrator Operation throws a TaskCanceledException after 100 seconds waiting for resources.
+            // This security exit clause was added to avoid a looping running forever, while resources are not appended.
+
+            // In this test, to simulate that scenario, a bundle operation is created expecting 10 resources, but only one is appended.
+            // The other 9 resources are never appened, forcing the looping to keep waiting for the remaining resources.
+            // This security exit clause will be activated after 120 seconds, cancelling the operation.
+
+            const int numberOfResources = 10;
+            const int maxWaitingTimeInSeconds = 150;
```

Comment:
nit: Is this supposed to be 120?

Reply:
I set it to 150 because it's supposed to be higher than 100s. 
The reason is: the internal cancellation will happen at 100s, if the external cancellation happens with 150, then it means that the internal one did not happen. I just added some additional time to the external cancellation. 

By the way, I just noticed that I need to update the comment with 100, instead of 120. 
I'll also decrease the external cancellation from 150s to 120s. It does not need to be as high as 150s.